atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolImpl.java

```diff
@@ -409,18 +409,18 @@ void setServersInPoolTo(ImmutableMap<CassandraServer, CassandraServerOrigin> des
         Map<CassandraServer, CassandraServerOrigin> serversToAdd = EntryStream.of(desiredServers)
                 .removeKeys(currentServers::contains)
                 .toImmutableMap();
-        SetView<CassandraServer> absentServers = Sets.difference(currentServers, desiredServers.keySet());
 
+        Set<CassandraServer> validatedServersToAdd =
+                validateNewHostsTopologiesAndMaybeAddToPool(getCurrentPools(), serversToAdd);
+
+        SetView<CassandraServer> absentServers = Sets.difference(currentServers, desiredServers.keySet());
```

Comment:
👍 The main original concern was that this might cause spurious failed validations e.g. with IP rotations. But we can always go back to the config (fast if need be), so that should be fine.